src/main/haskell/kore/test/Test/Terse.hs

```diff
@@ -199,6 +207,81 @@ f_2_expected
 f_2_expected f tuple expected =
   f_2_expected_name f tuple expected "f_2_expected with no name"
 
+
+-- $wrappedFunctions
+--
+-- Domain-specific function builders for code that consumes IO-wrapped
+--  resources. Tests are expected to look something like this:
+-- > testGroup "Combinations with operators that produce top or bottom"
+-- > [ mkEquals_ _True  _False `becomes` bottom
+-- >   ...
+-- > ]
+-- > where
+-- >   becomes makerInput =
+-- >     wrapped_maker_expected
+-- >       withSolver
+-- >       (\solver -> evaluateWith solver makerInput)
+--
+-- In `becomes`, a tree structure (the result of `mkEquals`, named
+-- `makerInput`) is given to the `solver` which produces a value
+-- (wrapped in `IO`) that is to be compared to an expected value.
+--
+-- In the previous example, `withSolver` is an
+-- `((IO a -> TestTree) -> TestTree)`
+-- that performs setup and teardown for each test. It is constructed with
+-- the test framework's `withResource`. Like this:
+-- >
+-- > import Test.Tasty
+-- >
+-- > withSolver :: (IO (MVar Solver) -> TestTree) -> TestTree
+-- > withSolver = withResource new free
+-- >   where
+-- >     new = SMT.newSolver SMT.defaultConfig
+-- >     free = SMT.stopSolver
+-- >
+-- >
+-- > The second argument is a function that "applies" the resource to
+-- > the test input to produce the value to be tested.
+
+wrapped_maker_expected_name_intention
+  :: (HasCallStack, Eq b, Show b, EqualWithExplanation b)
+  =>  ((IO a -> TestTree) -> TestTree)
+  -> (a -> IO b)  -- ^ take raw input, produce value to be checked
+  -> b            -- ^ the expected value
+  -> String       -- ^ the name of the generated test case
+  -> String       -- ^ a description of the intention of the comparison
+  -> TestTree
+wrapped_maker_expected_name_intention wrapper maker expected name intention =
+    wrapper $ \getResource ->
+        testCase name $ do
+            resource <- getResource
+            maker resource >>= assertEqualWithExplanation intention expected
+
+wrapped_maker_expected_name
+  :: (HasCallStack, Eq b, Show b, EqualWithExplanation b)
```

Comment:
The style guide requires 4-space indentation except in some specific cases (e.g. the "where" keyword). The list of allowed cases may not be comprehensive, and our code contains indentation levels that do not fit these requirements, but which make sense in context. However, most of the two-spaces indentation in this pull request seems to be used for normal-looking code, and it does not seem to me that it would be worth an exception from the style guide.

Reply:
Done.